zowe/zowe-cli
# feat(pkg): Add secrets SDK and `keyring` module

**What It Does**

This adds [`keytar-rs`](https://github.com/traeok/keytar-rs) into the CLI as a new package: `@zowe/secrets-for-zowe-sdk`. It also removes references to the old `node-keytar` library as it is no longer used.

- [x] Added the secrets SDK as a new package to the `zowe-cli` monorepo
- [x] Added `keyring` module to said secrets SDK
- [x] Added test and publish stages for secrets SDK to prepare for publishing

**How to Test**

**Note:** The test stages will fail - 2 integration tests are failing on the daemon because Imperative is not using the Secrets SDK yet. Once the Imperative branch is merged into the codebase, CLI will pass with the new version.

Reviewers should be able to clone the repo, `cd` into `packages/secrets`, and run `npm install` to build from source. In addition, reviewers should be able to run `npm run test` in this folder to verify that all tests are passing for the `keyring` module.

**Review Checklist**
I certify that I have:


**Additional Comments**

Sayonara, `node-keytar` :wave:
